test/Icicle/Test/Core/Arbitrary.hs

```diff
@@ -394,12 +441,12 @@ programForStreamType streamType
         (env', rs) <- gen_reduces sE (Map.insert nm (FunT [] t) pE) (n-1)
         return (env', (nm, red) : rs)
 
-  -- A reduction is either a fold or a latest
+  -- A reduction is a fold
   gen_reduce sE pE
    = do (i,t) <- oneof $ fmap return $ Map.toList sE
         at <- arbitrary
-        kx <- gen_exp (FunT [FunT [] at, FunT [] t] at) pE
-        zx <- gen_exp (FunT [] at) pE
+        kx <- gen_exp_for_type (FunT [FunT [] at, FunT [] t] at) pE
+        zx <- gen_exp_for_type (FunT [] at) pE
```

Comment:
just a suggestion: I think here, instead of generating the type at, you might be able to generate (zx, at) from tryExp?